src/effects/VST3/internal/ParameterChanges.h

```diff
@@ -0,0 +1,48 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file ParameterChanges.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#pragma once
+
+#include <vector>
+
+#include <pluginterfaces/vst/ivstparameterchanges.h>
+#include "ParameterValueQueue.h"
+#include "ParameterQueuePool.h"
+
+namespace internal
+{
+   //Implements IParameterChanges, uses ParameterQueuePool internally
+   //to avoid unnecessary allocations. Though both getParameterData
+   //and addParameterData, it's not recommended to store pointers to
```

Comment:
I think there is an incomplete clause in that comment

Reply:
It should be "Though both getParameterData and addParameterData return pointers to shared objects", but now I have doubts that such restriction is correct since VST plugins are allowed to access those objects as well